core/server/worker/src/main/java/alluxio/worker/netty/BlockReadHandler.java

```diff
@@ -101,21 +103,38 @@ protected DataBuffer getDataBuffer(BlockReadRequestContext context, Channel chan
       openBlock(context, channel);
       BlockReader blockReader = context.getBlockReader();
       Preconditions.checkState(blockReader != null);
-      if (mTransferType == FileTransferType.TRANSFER
-          && (blockReader instanceof LocalFileBlockReader)) {
+      DataBuffer dataBuffer = null;
+      if (mTransferType == FileTransferType.TRANSFER) {
+        dataBuffer = createDataFileChannel(blockReader, offset, len);
+      }
+      if (dataBuffer != null) {
+        return dataBuffer;
+      }
+      // create a non-zero-copy DataBuffer
+      ByteBuf buf = channel.alloc().buffer(len, len);
+      try {
+        while (buf.writableBytes() > 0 && blockReader.transferTo(buf) != -1) {
+        }
+        return new NettyDataBuffer(buf);
+      } catch (Throwable e) {
+        buf.release();
+        throw e;
+      }
+    }
+
+    private DataBuffer createDataFileChannel(BlockReader blockReader, long offset, long len) {
+      if (blockReader instanceof LocalFileBlockReader) {
         return new DataFileChannel(new File(((LocalFileBlockReader) blockReader).getFilePath()),
             offset, len);
-      } else {
-        ByteBuf buf = channel.alloc().buffer(len, len);
-        try {
-          while (buf.writableBytes() > 0 && blockReader.transferTo(buf) != -1) {
-          }
-          return new NettyDataBuffer(buf);
-        } catch (Throwable e) {
-          buf.release();
-          throw e;
+      }
+      if (blockReader instanceof DelegatingBlockReader) {
+        BlockReader delegatingBlockReader = ((DelegatingBlockReader) blockReader).getDelegate();
+        if (delegatingBlockReader instanceof StoreBlockReader) {
```

Comment:
Since `StoreBlockReader` is a subclass of `LocalFileBlockReader`, it's not necessary to check for `delegatingBlockReader instanceof StoreBlockReader` separately.